packages/node/src/login/oidc/ClientRegistrar.spec.ts

```diff
@@ -41,23 +48,239 @@ describe("ClientRegistrar", () => {
   }
 
   describe("getClient", () => {
-    it("isn't implemented", async () => {
+    it("Fails if there is not registration endpoint", async () => {
+      // Sets up the mock-up for DCR
+      const { Issuer } = jest.requireMock("openid-client");
+      const mockedIssuerConfig = mockIssuerConfig({
+        registration_endpoint: undefined,
+      });
+      const mockedIssuer = {
+        metadata: mockedIssuerConfig,
+      };
+      Issuer.discover = jest.fn().mockResolvedValueOnce(mockedIssuer);
+
+      // Run the test
       const clientRegistrar = getClientRegistrar({
         storage: mockStorageUtility({}),
       });
-      const registrationUrl = "https://idp.com/register";
       await expect(
         clientRegistrar.getClient(
           {
             sessionId: "mySession",
             redirectUrl: "https://example.com",
           },
-          {
-            ...IssuerConfigFetcherFetchConfigResponse,
-            registrationEndpoint: registrationUrl,
-          }
+          IssuerConfigFetcherFetchConfigResponse
         )
-      ).rejects.toThrow();
+      ).rejects.toThrow(
+        "Dynamic client registration cannot be performed, because issuer does not have a registration endpoint"
```

Comment:
Can we make this message a little more helpful by providing the Issuer URL or something, e.g. something like (just to give you the idea):
```suggestion
        "Dynamic client registration cannot be performed, because issuer at [${issuer}] does not have a registration endpoint (expected a valid HTTP response from something like [${issuer}/register])."
```

Reply:
The error message does include additional information (``Dynamic client registration cannot be performed, because issuer does not have a registration endpoint: ${JSON.stringify(issuer.metadata)}``), but I omitted that part of the error message in the test to make it easier to read.